src/components/Header/index.web.js

```diff
@@ -0,0 +1,102 @@
+import React, {PureComponent, Fragment} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {Image, Text, View} from 'react-native';
+import {isEmpty} from 'lodash';
+
+import stylePropType from '../../util/stylePropType';
+import routePropType from '../../util/routePropType';
+import NavItem from './NavItem';
+import SessionService from '../../services/session';
+import {requestFetchToken} from '../../actions/session';
+
+import styles from './styles';
+
+const brandImageDefault = require('../../images/brand.png');
+
+class Header extends PureComponent {
+    static propTypes = {
+        brandImage: PropTypes.oneOfType([
+            PropTypes.number,
+            PropTypes.string
+        ]),
+        rightImage: PropTypes.oneOfType([
+            PropTypes.number,
+            PropTypes.string
+        ]),
+        style: stylePropType,
+        text: PropTypes.string,
+        routes: PropTypes.arrayOf(routePropType).isRequired,
+        token: PropTypes.string,
+        version: PropTypes.string,
+        requestFetchToken: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+        brandImage: brandImageDefault,
+        style: {},
+        token: null,
+        text: null,
+        rightImage: null,
+        version: null
+    };
+
+    async signOut(closeSession) {
+        if (closeSession) {
+            await SessionService.clearSession();
+            this.props.requestFetchToken();
+        }
+    }
+
+    render() {
+        const {
+            routes, text, brandImage, style, token, rightImage, version
+        } = this.props;
+
+        return (
+            <Fragment>
```

Comment:
creo que hay bastante codigo duplicado entre este componente y el `index.js`, creo que estaria bueno separarlo primero por componentes mas chicos ejemplo la version por un lado, las imagenes por otro (fijate este PR como ejemplo https://github.com/indec-it/react-native-commons/pull/71) y en otro componente que sea la union de todo eso y luego aca y en el `index.js` llamas a ese componente